src/instrument/devices/registers_device.py

```diff
@@ -0,0 +1,24 @@
+"""
+EPICS PVs as Storage registers.
+"""
+
+from ophyd import Component
+from ophyd import Device
+from ophyd import EpicsSignal
+
+
+class EpicsPvStorageRegisters(Device):
+    measurement_num = Component(EpicsSignal, "StrReg21", string=True)
+    qnw_index = Component(EpicsSignal, "StrReg22", string=True)
+    qmap_file = Component(EpicsSignal, "StrReg23", string=True)
+    image_file_name = Component(EpicsSignal, "StrReg24", string=True)
+    experiment_name = Component(EpicsSignal, "StrReg25", string=True)
+    cycle_name = Component(EpicsSignal, "StrReg26", string=True)
+    workflow_name = Component(EpicsSignal, "StrReg27", string=True)
+    analysis_machine = Component(EpicsSignal, "StrReg29", string=True)
+    # TODO StrReg28?
+    # metadata_file = Component(EpicsSignal, "StrReg27", string=True)  # FIXME: same as workflow_name

```

Comment:
Which register should be used for the metadata file name?  The PyEpics code used StrReg27 but that is used for the DM workflow name.

Reply:
You'll pick StrRef28 for the `metadata_file`, then?